server/index.js

```diff
@@ -179,6 +183,9 @@ app.get('/api/myprofile', (req, res, next) => {
 
 app.get('/api/username', (req, res, next) => {
   const { userId } = req.user;
+  if (!userId) {
+    throw new ClientError(400, 'userId is a required filed');
+  }
```

Comment:
You can remove these checks, because if there was no token, the `authorizationMiddleware` would not let this code run.